packages/typespec-test/test/Keyvault.Management/generated/typespec-ts/src/models/models.ts

```diff
@@ -0,0 +1,3637 @@
+  readonly tags?: Record<string, string>;
+  /** Purge protection status of the original vault. */
+  readonly purgeProtectionEnabled?: boolean;
+}
+
+export function deletedVaultPropertiesDeserializer(
+  item: any,
+): DeletedVaultProperties {
+  return {
+    vaultId: item["vaultId"],
+    location: item["location"],
+    deletionDate: !item["deletionDate"]
+      ? item["deletionDate"]
+      : new Date(item["deletionDate"]),
+    scheduledPurgeDate: !item["scheduledPurgeDate"]
+      ? item["scheduledPurgeDate"]
+      : new Date(item["scheduledPurgeDate"]),
+    tags: item["tags"],
+    purgeProtectionEnabled: item["purgeProtectionEnabled"],
+  };
+}
+
+/** The resource model definition for a Azure Resource Manager proxy resource. It will not have tags and a location */
+export interface ProxyResource extends Resource {}
+
+export function proxyResourceSerializer(item: ProxyResource): any {
+  return item;
+}
+
+export function proxyResourceDeserializer(item: any): ProxyResource {
+  return {
+    id: item["id"],
+    name: item["name"],
+    type: item["type"],
+    systemData: !item["systemData"]
+      ? item["systemData"]
+      : systemDataDeserializer(item["systemData"]),
+  };
+}
+
+/** The response of a DeletedVault list operation. */
+export interface _DeletedVaultListResult {
+  /** The DeletedVault items on this page */
+  value: DeletedVault[];
+  /** The link to the next page of items */
+  nextLink?: string;
+}
+
+export function _deletedVaultListResultDeserializer(
+  item: any,
+): _DeletedVaultListResult {
+  return {
+    value: deletedVaultArrayDeserializer(item["value"]),
+    nextLink: item["nextLink"],
+  };
+}
+
+export function deletedVaultArrayDeserializer(
+  result: Array<DeletedVault>,
+): any[] {
+  return result.map((item) => {
+    return deletedVaultDeserializer(item);
+  });
+}
+
+/** The response of a Resource list operation. */
+export interface _ResourceListResult {
+  /** The Resource items on this page */
+  value: Resource_1[];
+  /** The link to the next page of items */
+  nextLink?: string;
+}
+
+export function _resourceListResultDeserializer(
+  item: any,
+): _ResourceListResult {
+  return {
+    value: resourceArrayDeserializer(item["value"]),
+    nextLink: item["nextLink"],
+  };
+}
+
+export function resourceArrayDeserializer(result: Array<Resource>): any[] {
+  return result.map((item) => {
+    return resourceDeserializer_1(item);
+  });
+}
+
+/** Key Vault resource */
+export interface Resource_1 {
+  /** Fully qualified identifier of the key vault resource. */
+  readonly id?: string;
+  /** Name of the key vault resource. */
+  readonly name?: string;
+  /** Resource type of the key vault resource. */
+  readonly type?: string;
+  /** Azure location of the key vault resource. */
+  readonly location?: string;
+  /** Tags assigned to the key vault resource. */
+  readonly tags?: Record<string, string>;
+}
+
+export function resourceDeserializer_1(item: any): Resource_1 {
```

Comment:
Because the model `Resource` is duplicated, `Resource_1` and the corresponding function Deserializer_1 are generated. Update the codegen logic to not export such similar functions.

Reply:
Is it becuase the Serializer_1 is not matching our original condition so it is wronly exposed?

Reply:
yes